Deep_Learning/HoneypotAI/tests/test_main.py

```
@@ -0,0 +1,441 @@
+        mock_detector.return_value = mock_detector_instance
+        mock_trainer.return_value = mock_trainer_instance
+        mock_response.return_value = mock_response_instance
+        
+        app = HoneypotAI()
+        
+        # Components should be initialized
+        assert app.honeypot_manager == mock_manager_instance
+        assert app.threat_detector == mock_detector_instance
+        assert app.online_trainer == mock_trainer_instance
+        assert app.adaptive_response == mock_response_instance
+        assert app.running == False
```

Ruff (0.12.2)

Avoid equality comparisons to `False`; use `not app.running:` for false checks

Replace with `not app.running`

(E712)